### ISSUE TITLE
Migration 2.10 -> 2.13.1 drops many comments, mangles some

### ISSUE THREAD

**Issue by stanislav-brabec** [WeblateOrg/weblate#1496]
After migration from 2.10 to 2.13.1, thousands comments are lost, and some comments changed its text.

It seems that changed comments are related to #1494.

Quick look into the diff indicates that all lost comments have one of table columns equal to NULL, all survived have this column non-zero.

Comments were created in weblate-2.5 and weblate-2.6, then migrated to 2.10.

### Steps to reproduce
1. Dump database before migration (in my case it was mysql) to weblate210.sql
2. ./manage.py migrate
3. Dump database before migration (in my case it was mysql) to weblate2131.sql
4. Perform commands to compare suggestions:
```shell
grep 'INSERT_INTO `trans_comment`' weblate210.sql >before.cut
grep 'INSERT_INTO `trans_comment`' weblate2131.sql >after.cut
sed "s/),(/),\\n(/g" <before.cut >before.lines
sed "s/),(/),\\n(/g" <after.cut >after.lines
sed 's/^\(([0-9]*,\)'\''[0-9a-f]*'\'',/\1/' <before.lines >before.comp
sed 's/,[^,()]*)/)/' <after.lines >after.comp
diff -u before.comp after.comp
```

### Actual behaviour
```diff
 (2370,'Suggested translation','2016-07-08 12:04:50',286,30,190),
 (2372,'Seems to be a name (not translatable)','2016-07-08 12:11:13',286,30,190),
 (2374,'Probably product name (not translatable) — see: https://www.suse.com/products/suse-enterprise-storage','2016-07-08 12:14:09',286,30,190),
-(2376,'Style enhancement: \"non voluto/i\" changed to \"indesiderato/i\"','2016-07-08 12:15:52',NULL,34,168),
-(2378,'Style enhancement','2016-07-08 13:16:31',NULL,34,168),
-(2380,'Style issue','2016-07-08 13:38:26',NULL,30,168),
-(2382,'Style enhancement','2016-07-08 13:39:50',NULL,30,168),
 (2384,'Corrected translation','2016-07-08 14:13:55',286,34,190),
 (2386,'Corrected translation','2016-07-08 14:25:10',286,34,190),
 (2388,'Corrected translation','2016-07-08 14:29:16',286,34,190),
```
(Thousands comments are dropped.)

and elsewhere
```diff
(1124,'\"Towards\" means \"向こう側\", not \"現実化\". \"現実化\" is \"realize\"or similar words.','2016-07-04 11:18:27',256,30,18),
 (1126,'\"Modular\" is often translated to \"モジュール型\", it is more commonly used.','2016-07-04 11:20:32',256,30,18),
 (1128,'This is a slideshow title message, so it is better that message ends with noun, not verb.','2016-07-04 11:23:43',256,30,18),
-(1130,'Original translation wrong (incomplete, terminology).','2016-07-04 11:25:22',286,34,234),
+(1130,'Original translation wrong (incomplete).','2016-07-04 11:25:22',286,34,234,-360503172539850206),
 (1132,'Suggestion was better.','2016-07-04 11:29:39',286,34,234),
 (1134,'Suggestion was better.','2016-07-04 11:29:54',286,34,234),
 (1136,'Better phrase used in suggestion.','2016-07-04 11:47:34',286,34,234),
```

### Expected behaviour
Not mangle comments.

### Server configuration
 * Weblate 2.13.1
 * Python 2.7.12
 * Django 1.10.5
 * six 1.9.0
 * social-auth-core 1.2.0
 * social-auth-app-django 1.1.0
 * django-appconf 1.0.1
 * Translate Toolkit 2.0.0
 * Whoosh 2.7.0
 * defusedxml 0.4.1
 * Git 2.12.0
 * Pillow (PIL) 1.1.7
 * dateutil 2.4.2
 * lxml 3.6.1
 * django-crispy-forms 1.6.1
 * compressor 2.1.1
 * djangorestframework 3.5.3
 * pytz 2015.2
 * pyuca N/A
 * PyYAML 3.11
 * Mercurial 3.8.3
 * git-svn 2.12.0
 * git-review 1.25.0
 * Database backends: django.db.backends.mysql

**Comment by nijel** [comment]
For the line with comma, it's same as in https://github.com/WeblateOrg/weblate/issues/1494#issuecomment-302711526 - it has been changed by your sed.

Can you please check what column was NULL on those removed entries?

**Comment by nijel** [comment]
I've found that anyway - these are source scope comments and they were indeed not migrated properly.

**Comment by stanislav-brabec** [comment]
I can confirm that the string mangling was caused by an incorrect sed expression.

I will apply your patch and re-run the migration. I will post the result in few days. (The migration takes many hours.)

**Comment by nijel** [comment]
There was another fix to that patch in 80ed96d (but that would fail very early).

**Comment by stanislav-brabec** [comment]
Confirming that 2.13.1 + 1c7cd65 + 74ade3f + 80ed96d correctly migrates all comments.
